### ISSUE TITLE
Container not transitive ?

### ISSUE THREAD

**Issue by kildirian** [MyreMylar/pygame_gui#97]
Hi,
I try to plot an UIElement C contained in an UIElement B contained in an UIElement A.
Each time I set the container (the parent) as parameter with relative_rect relative to its parents.
But doing this, the UIElement C is display relatively to B but not to A.

For the sake of clarity, I take an example:
if topleft A is in (100,100)
B relative_topleft to A is (50,50) (so 150,150 regarding the screen)
C relative_topleft to B is (25,25)
Then I get a C element plot at (75,75) not at (175,175).

Why is it not transitive ? should I compute myself the difference ? Or is there a way to do it ?

Here is my code, there are some french words sorry for that.

```

class MenuUnite():
    def __init__(self,manager):
        self.manager=manager
        
    def start(self,incoming_data):
        unite=incoming_data["unite"]
        player=incoming_data["player"]
        
    
        self.background=UIPanel(pygame.Rect(Config.TOPLEFT_MENU,(Config.LARGEUR_MENU,Config.HAUTEUR_MENU)),
                                manager=self.manager,
                                starting_layer_height=0,
                                object_id="#menuPanel")
        
        
        
        x=Config.LARGEUR_MENU/10
        y=Config.HAUTEUR_MENU/10
        
        self.label=UILabel(pygame.Rect(x,y/2,8*x,y),
                           text="Menu Informations",
                           manager=self.manager,
                           container=self.background,
                           object_id="#title")
        
        
        self.panelInfosPlayer=UIPanelInfosPlayer(pygame.Rect(x,2*y,8*x,3*y),
                                               player=player,
                                               manager=self.manager,
                                               container=self.background,
                                               object_id="#sub_panel")


             
class UIPanelInfosPlayer(UIPanel):
    def __init__(self,relative_rect,player,manager,container,object_id):
        UIPanel.__init__(self,relative_rect=relative_rect,manager=manager,container=container,object_id=object_id,starting_layer_height=0)
        infos=player.getInfos()
        
        
        x=self.rect.size[0]/10
        y=self.rect.size[1]/10
        
        self.label=UILabel(pygame.Rect(2*x,0,6*x,2*y),
                           text='Informations Joueur',
                           manager=manager,
                           container=self)
        
        self.elements=[]
        keys=["Argents","Rentes","Unites","Batiments"]
        for i in range(len(keys)):
            label=UILabel(pygame.Rect(x,3*y+i*1.5*y*y,4*x,y),
                           text=keys[i]+": "+str(infos[keys[i]]),
                           manager=manager,
                           container=self)
            self.elements.append(label)
            
            
        avatarImage=pygame.transform.scale(player.avatar,(int(4*x),int(4*y)))
        self.avatar=UIImage(pygame.Rect(5.5*x,2*y,4*x,4*y),
                            image_surface=avatarImage,
                            manager=manager,
                            container=self)
        
        self.label=UILabel(pygame.Rect(5.5*x,7*y,4*x,y),
                   text=player.name,
                   manager=manager,
                   container=self)
                           
                        
        color=pygame.Surface((int(3*x),int(y)))
        color.fill(player.color_team)
        self.colorBar=UIImage(pygame.Rect((5.5*x,2*y),color.get_rect().size),
                            image_surface=color,
                            manager=manager,
                            container=self)
```

**Comment by MyreMylar** [comment]
The elements should be relative all the way down the line in general but I believe there is a bug in the released (0.5.1) version of the UIPanel element currently fixed in the master branch.

This should be fixed when I do the next release 0.5.5, if you want a fix now you could just copy the current ui_panel.py file from the master branch here:

https://github.com/MyreMylar/pygame_gui/tree/master/pygame_gui/elements

Over to your installed version of pygame_gui (if you know how that is done).

**Comment by kildirian** [comment]
There are some errors of import.
ImportError: cannot import name 'IUIContainerInterface' from 'pygame_gui.core.interfaces'
I didn't find the IUIContainerInterface in your master branch

**Comment by MyreMylar** [comment]
Ah yes, I guess I forgot I added some new interfaces too. Glad you got it working!

**Comment by MyreMylar** [comment]
This should be fixed now in 0.5.5
